pkg/pool-weighted/contracts/managed/ManagedPool.sol

```diff
@@ -225,29 +263,35 @@ contract ManagedPool is ManagedPoolSettings {
         uint256[] memory balances,
         bytes memory userData
     ) internal virtual override returns (uint256 bptAmountOut, uint256[] memory amountsIn) {
-        (IERC20[] memory tokens, ) = _getPoolTokens();
+        // The Vault passes an array of balances which includes the pool's BPT (This always sits in the first position).
+        // We want to separate this from the other balances before continuing with the join.
+        uint256 virtualSupply;
+        (virtualSupply, balances) = ComposablePoolLib.dropBptFromBalances(totalSupply(), balances);
 
+        (IERC20[] memory tokens, ) = _getPoolTokens();
         uint256[] memory scalingFactors = _scalingFactors(tokens);
         _upscaleArray(balances, scalingFactors);
 
         // The AUM fee calculation is based on inflating the Pool's BPT supply by a target rate.
         // We then must collect AUM fees whenever joining or exiting the pool to ensure that LPs only pay AUM fees
         // for the period during which they are an LP within the pool: otherwise an LP could shift their share of the
         // AUM fees onto the remaining LPs in the pool by exiting before they were paid.
-        uint256 preFeeCollectionSupply = _getVirtualSupply();
-        uint256 postFeeCollectionSupply = preFeeCollectionSupply + _collectAumManagementFees(preFeeCollectionSupply);
+        virtualSupply += _collectAumManagementFees(virtualSupply);
```

Comment:
I'd consider renaming this to `actualSupply` (or rather, creating a new variable)

Reply:
Well it is technically the virtual supply at this point (as the BPT has been minted). We should have stack space for this though.